### ISSUE TITLE
Using StartMaja with MAJA v3.2.2

### ISSUE THREAD

**Issue by c00kiemon5ter** [CNES/Start-MAJA#82]
tl;dr: Is there a version of StartMaja that we can use to drive MAJA v3.2.2?

---

Hello,

due to the issues we have encountered with MAJA v4.2.1 (described in #81) we are falling back to MAJA v3.2.2. However, with that (closed source) version there is no StartMaja bundled. AFAIK, StartMaja was its own project (https://github.com/CNES/Start-MAJA) that was merged into the MAJA codebase, before it was open-sourced. 

I tried using StartMaja as found in this repository, but have it invoke MAJA 3.2.2. However, that failed as the configuration files are different (userconf schema validation failure). I dropped that path and tried to find whether an old repo is still around.

I have come across [@olivierhagolle's repo of StartMaja][ostartmaja]. Using the `master` branch (`v3.1` did not work), I failed again to make it work with MAJA v3.2.2. I supply the configuration in the needed format, [the needed GIPP param files][gipp] (and specifically `GIPP_MAJA_3_1_S2AS2B_MUSCATE_TM` plus the fetched content from zenodo), and the right CLI params. I invoke the process but it fails. Following is the invocation command and resulting logs. Note that this version of StartMaja needs python2 and that I supply an empty value for the site name to avoid re-organizing/recreating the L1C products (a read-only repository).

```bash
$ cat run.sh 
#!/bin/sh

PATH="${MAJA_DIR}/bin:${MINICONDA_DIR}/bin:${PATH}"
. "${MINICONDA_DIR}/etc/profile.d/conda.sh"

conda activate maja
python maja/source/start_maja.py -f config.txt -t 34SFJ -d 20190301 -e 20190801 -s "" -c fix

$ bash ./run.sh
2020-12-20T19:57:03.585426 13fa8c657432 maja-processing-3.2.2 3.2 [000000170640] [I] MAJA Software Version : 3.2.2
2020-12-20T19:57:03.586117 13fa8c657432 maja-processing-3.2.2 3.2 [000000170640] [I] Overloads the conf value in the command line by '/root/work//34SFJ/fix//userconf'.
2020-12-20T19:57:03.592302 13fa8c657432 maja-processing-3.2.2 3.2 [000000170640] [I] -------------------------------------------------------------------------
2020-12-20T19:57:03.592407 13fa8c657432 maja-processing-3.2.2 3.2 [000000170640] [I] Configuration User System parameters:
2020-12-20T19:57:03.592519 13fa8c657432 maja-processing-3.2.2 3.2 [000000170640] [I]     *      L2Note                                          : L2 note
2020-12-20T19:57:03.592598 13fa8c657432 maja-processing-3.2.2 3.2 [000000170640] [I]     *      L3Note                                          : L3 note
2020-12-20T19:57:03.592682 13fa8c657432 maja-processing-3.2.2 3.2 [000000170640] [I]     *      CheckToolsNote                                  : Checktool note
2020-12-20T19:57:03.592788 13fa8c657432 maja-processing-3.2.2 3.2 [000000170640] [I]     *      NbThreads                                       : 8
2020-12-20T19:57:03.592882 13fa8c657432 maja-processing-3.2.2 3.2 [000000170640] [I]     *      EnableCleaningTemporaryDirectory                : true
2020-12-20T19:57:03.592969 13fa8c657432 maja-processing-3.2.2 3.2 [000000170640] [I]     *      EnableCachingConvertReflectanceData             : true
2020-12-20T19:57:03.593056 13fa8c657432 maja-processing-3.2.2 3.2 [000000170640] [I]     *      EnableCleaningCachingDirectoryBeforeProcessing  : false
2020-12-20T19:57:03.593146 13fa8c657432 maja-processing-3.2.2 3.2 [000000170640] [I]     *      EnableCleaningCachingDirectoryAfterProcessing   : true
2020-12-20T19:57:03.593233 13fa8c657432 maja-processing-3.2.2 3.2 [000000170640] [I]     *      EnableL2ResolutionProcessing                    : true
2020-12-20T19:57:03.593322 13fa8c657432 maja-processing-3.2.2 3.2 [000000170640] [I]     *      PreProcessingCleanInputZipFiles                 : false
2020-12-20T19:57:03.593413 13fa8c657432 maja-processing-3.2.2 3.2 [000000170640] [I]     *      PostProcessingZipFiles                          : false
2020-12-20T19:57:03.593501 13fa8c657432 maja-processing-3.2.2 3.2 [000000170640] [I]     *      PostProcessingCleanFiles                        : false
2020-12-20T19:57:03.593601 13fa8c657432 maja-processing-3.2.2 3.2 [000000170640] [I]     *      CheckXMLFilesWithSchema                         : false
2020-12-20T19:57:03.593696 13fa8c657432 maja-processing-3.2.2 3.2 [000000170640] [I] -------------------------------------------------------------------------
2020-12-20T19:57:03.593793 13fa8c657432 maja-processing-3.2.2 3.2 [000000170640] [I] Using default Admin configuration system file.
2020-12-20T19:57:03.598901 13fa8c657432 maja-processing-3.2.2 3.2 [000000170640] [I] -------------------------------------------------------------------------
2020-12-20T19:57:03.599037 13fa8c657432 maja-processing-3.2.2 3.2 [000000170640] [I] Configuration Admin System parameters:
2020-12-20T19:57:03.599181 13fa8c657432 maja-processing-3.2.2 3.2 [000000170640] [I]     *      System                                          : MAJA
2020-12-20T19:57:03.599311 13fa8c657432 maja-processing-3.2.2 3.2 [000000170640] [I]     *      Tolerances / Footprint / Coordinate             : 1e-06
2020-12-20T19:57:03.599417 13fa8c657432 maja-processing-3.2.2 3.2 [000000170640] [I]     *           ''          ''    / Direction              : 1e-06
2020-12-20T19:57:03.599526 13fa8c657432 maja-processing-3.2.2 3.2 [000000170640] [I]     *      EnablePerformanceMeasureAlgorithms              : false
2020-12-20T19:57:03.599631 13fa8c657432 maja-processing-3.2.2 3.2 [000000170640] [I] -------------------------------------------------------------------------
2020-12-20T19:57:03.599796 13fa8c657432 maja-processing-3.2.2 3.2 [000000170640] [I] ******************************************************************************
2020-12-20T19:57:03.599918 13fa8c657432 maja-processing-3.2.2 3.2 [000000170640] [I] Application information: 
2020-12-20T19:57:03.600026 13fa8c657432 maja-processing-3.2.2 3.2 [000000170640] [I]     *      Processor Name               : MAJA_L2_BACKWARD_CHAIN
2020-12-20T19:57:03.600129 13fa8c657432 maja-processing-3.2.2 3.2 [000000170640] [I]     *      Test mode                    : false
2020-12-20T19:57:03.600245 13fa8c657432 maja-processing-3.2.2 3.2 [000000170640] [I]     *      UserConfigSystemFilename     : /root/L2A//34SFJ/fix//.maja-working-directory/MAJAUserConfigSystem.xml
2020-12-20T19:57:03.600356 13fa8c657432 maja-processing-3.2.2 3.2 [000000170640] [I]     *      AdminConfigSystemFileName    : /root/L2A//34SFJ/fix//.maja-working-directory/MAJAAdminConfigSystem.xml
2020-12-20T19:57:03.600481 13fa8c657432 maja-processing-3.2.2 3.2 [000000170640] [I]     *      Specifics command line parameters:
2020-12-20T19:57:03.600594 13fa8c657432 maja-processing-3.2.2 3.2 [000000170640] [I]           - TileId: 34SFJ
2020-12-20T19:57:03.600736 13fa8c657432 maja-processing-3.2.2 3.2 [000000170640] [I] ******************************************************************************
2020-12-20T19:57:03.600895 13fa8c657432 maja-processing-3.2.2 3.2 [000000170640] [I] The process uses 8 thread(s). Note: possible to use of maximum 128 threads.
2020-12-20T19:57:03.601119 13fa8c657432 maja-processing-3.2.2 3.2 [000000170640] [I] Starting L2BackwardProcessor::PreProcessing() ...
2020-12-20T19:57:03.685151 13fa8c657432 maja-processing-3.2.2 3.2 [000000170640] [I] There are 8 L1 images products.
2020-12-20T19:57:03.685317 13fa8c657432 maja-processing-3.2.2 3.2 [000000170640] [I]  - /root/work//34SFJ/fix//in/S2B_MSIL1C_20190302T090909_N0207_R050_T34SFJ_20190302T115834.SAFE/MTD_MSIL1C.xml
2020-12-20T19:57:03.685519 13fa8c657432 maja-processing-3.2.2 3.2 [000000170640] [I]  - /root/work//34SFJ/fix//in/S2B_MSIL1C_20190305T092029_N0207_R093_T34SFJ_20190305T132100.SAFE/MTD_MSIL1C.xml
2020-12-20T19:57:03.685632 13fa8c657432 maja-processing-3.2.2 3.2 [000000170640] [I]  - /root/work//34SFJ/fix//in/S2A_MSIL1C_20190307T090821_N0207_R050_T34SFJ_20190307T104605.SAFE/MTD_MSIL1C.xml
2020-12-20T19:57:03.685749 13fa8c657432 maja-processing-3.2.2 3.2 [000000170640] [I]  - /root/work//34SFJ/fix//in/S2A_MSIL1C_20190310T092021_N0207_R093_T34SFJ_20190310T112345.SAFE/MTD_MSIL1C.xml
2020-12-20T19:57:03.685873 13fa8c657432 maja-processing-3.2.2 3.2 [000000170640] [I]  - /root/work//34SFJ/fix//in/S2A_MSIL1C_20190317T090721_N0207_R050_T34SFJ_20190317T130844.SAFE/MTD_MSIL1C.xml
2020-12-20T19:57:03.685997 13fa8c657432 maja-processing-3.2.2 3.2 [000000170640] [I]  - /root/work//34SFJ/fix//in/S2B_MSIL1C_20190322T090649_N0207_R050_T34SFJ_20190322T130915.SAFE/MTD_MSIL1C.xml
2020-12-20T19:57:03.686121 13fa8c657432 maja-processing-3.2.2 3.2 [000000170640] [I]  - /root/work//34SFJ/fix//in/S2B_MSIL1C_20190325T092029_N0207_R093_T34SFJ_20190325T121158.SAFE/MTD_MSIL1C.xml
2020-12-20T19:57:03.686246 13fa8c657432 maja-processing-3.2.2 3.2 [000000170640] [I]  - /root/work//34SFJ/fix//in/S2A_MSIL1C_20190330T092031_N0207_R093_T34SFJ_20190330T105736.SAFE/MTD_MSIL1C.xml
2020-12-20T19:57:03.754157 13fa8c657432 maja-processing-3.2.2 3.2 [000000170640] [P] -----------------------------------------------------------------------------------------------
2020-12-20T19:57:03.754316 13fa8c657432 maja-processing-3.2.2 3.2 [000000170640] [P] Starting L2Processor::PreProcessing() ...
2020-12-20T19:57:03.754540 13fa8c657432 maja-processing-3.2.2 3.2 [000000170640] [I] List of the 2 satellite(s) activated :
2020-12-20T19:57:03.754674 13fa8c657432 maja-processing-3.2.2 3.2 [000000170640] [I]  - SENTINEL-2A ('common' satellite SENTINEL-2_ and plugin SENTINEL2_TM)
2020-12-20T19:57:03.754810 13fa8c657432 maja-processing-3.2.2 3.2 [000000170640] [I]  - SENTINEL-2B ('common' satellite SENTINEL-2_ and plugin SENTINEL2_TM)
2020-12-20T19:57:03.754922 13fa8c657432 maja-processing-3.2.2 3.2 [000000170640] [I] Starting reading and checking input data common for each plugin activated...
2020-12-20T19:57:03.755057 13fa8c657432 maja-processing-3.2.2 3.2 [000000170640] [I] According to the input data, there is(are) 1 plugin(s) activated:
2020-12-20T19:57:03.755195 13fa8c657432 maja-processing-3.2.2 3.2 [000000170640] [I]  - SENTINEL2_TM
2020-12-20T19:57:03.755334 13fa8c657432 maja-processing-3.2.2 3.2 [000000170640] [I]  - SENTINEL2_TM
2020-12-20T19:57:03.756780 13fa8c657432 maja-processing-3.2.2 3.2 [000000170640] [I] Starting DEM reading data for the 'unique' (common) satellite 'SENTINEL-2_' ...
2020-12-20T19:57:03.756957 13fa8c657432 maja-processing-3.2.2 3.2 [000000170640] [I] UnCompress the DBL product </root/work/34SFJ/fix/in/S2__TEST_AUX_REFDE2_34SFJ_3001.DBL> ...
2020-12-20T19:57:03.757199 13fa8c657432 maja-processing-3.2.2 3.2 [000000170640] [I] The DBL product </root/work/34SFJ/fix/in/S2__TEST_AUX_REFDE2_34SFJ_3001.DBL> is already uncompressed (.DBL.DIR directory already exists).
2020-12-20T19:57:03.757600 13fa8c657432 maja-processing-3.2.2 3.2 [000000170640] [I] l_NumberOfL2Resolution : 2
2020-12-20T19:57:03.757725 13fa8c657432 maja-processing-3.2.2 3.2 [000000170640] [I] Starting DTM reading ...
2020-12-20T19:57:03.770980 13fa8c657432 maja-processing-3.2.2 3.2 [000000170640] [I] UnCompress the DBL product </root/work/34SFJ/fix/in/S2__TEST_AUX_REFDE2_34SFJ_3001.DBL> ...
2020-12-20T19:57:03.771234 13fa8c657432 maja-processing-3.2.2 3.2 [000000170640] [I] The DBL product </root/work/34SFJ/fix/in/S2__TEST_AUX_REFDE2_34SFJ_3001.DBL> is already uncompressed (.DBL.DIR directory already exists).
2020-12-20T19:57:03.771659 13fa8c657432 maja-processing-3.2.2 3.2 [000000170640] [I] Starting DTM reading (nb resolutions=2) ...
2020-12-20T19:57:07.503754 13fa8c657432 maja-processing-3.2.2 3.2 [000000170640] [I] In the input data directory, there is(are) 0 meteo filename (EXO_METDTA).
2020-12-20T19:57:07.504392 13fa8c657432 maja-processing-3.2.2 3.2 [000000170640] [I] In the input data directory, there is(are) 0 cams filename (EXO_CAMS).
2020-12-20T19:57:07.506151 13fa8c657432 maja-processing-3.2.2 3.2 [000000170640] [I] Starting reading and checking input data for all satellites activated...
2020-12-20T19:57:07.506636 13fa8c657432 maja-processing-3.2.2 3.2 [000000170640] [I] Initialize the Plugin instance detected with the Plugin Name 'SENTINEL2_TM' ...
2020-12-20T19:57:07.511810 13fa8c657432 maja-processing-3.2.2 3.2 [000000170640] [I] The number of streaming lines of buffer for reading/writing images is '800'.
2020-12-20T19:57:07.511968 13fa8c657432 maja-processing-3.2.2 3.2 [000000170640] [I] Using default Admin configuration camera file.
2020-12-20T19:57:07.516524 13fa8c657432 maja-processing-3.2.2 3.2 [000000170640] [I] Starting reading data for the satellite 'SENTINEL-2A' and the plugin used is 'SENTINEL2_TM' ...
2020-12-20T19:57:07.516728 13fa8c657432 maja-processing-3.2.2 3.2 [000000170640] [P] Starting GIP_L2SMAC reading ...
2020-12-20T19:57:07.518111 13fa8c657432 maja-processing-3.2.2 3.2 [000000170640] [P] Starting GIP_L2TOCR reading ...
2020-12-20T19:57:07.518975 13fa8c657432 maja-processing-3.2.2 3.2 [000000170640] [I] UnCompress the DBL product </root/work/34SFJ/fix/in/S2A_TEST_GIP_L2TOCR_L_CONTINEN_10005_20150703_21000101.DBL> ...
2020-12-20T19:57:07.519177 13fa8c657432 maja-processing-3.2.2 3.2 [000000170640] [I] The DBL product </root/work/34SFJ/fix/in/S2A_TEST_GIP_L2TOCR_L_CONTINEN_10005_20150703_21000101.DBL> is already uncompressed (.DBL.DIR directory already exists).
2020-12-20T19:57:07.520297 13fa8c657432 maja-processing-3.2.2 3.2 [000000170640] [P] Starting GIP_L2WATV reading ...
2020-12-20T19:57:07.521145 13fa8c657432 maja-processing-3.2.2 3.2 [000000170640] [I] UnCompress the DBL product </root/work/34SFJ/fix/in/S2A_TEST_GIP_L2WATV_L_CONTINEN_10005_20150703_21000101.DBL> ...
2020-12-20T19:57:07.521354 13fa8c657432 maja-processing-3.2.2 3.2 [000000170640] [I] The DBL product </root/work/34SFJ/fix/in/S2A_TEST_GIP_L2WATV_L_CONTINEN_10005_20150703_21000101.DBL> is already uncompressed (.DBL.DIR directory already exists).
2020-12-20T19:57:07.521716 13fa8c657432 maja-processing-3.2.2 3.2 [000000170640] [I] Allocating LookUpTable image dimension = 3
2020-12-20T19:57:07.522059 13fa8c657432 maja-processing-3.2.2 3.2 [000000170640] [P] Starting GIP_L2DIRT uncompressing ...
2020-12-20T19:57:07.522912 13fa8c657432 maja-processing-3.2.2 3.2 [000000170640] [I] UnCompress the DBL product </root/work/34SFJ/fix/in/S2A_TEST_GIP_L2DIRT_L_CONTINEN_10005_20150703_21000101.DBL> ...
2020-12-20T19:57:07.523129 13fa8c657432 maja-processing-3.2.2 3.2 [000000170640] [I] The DBL product </root/work/34SFJ/fix/in/S2A_TEST_GIP_L2DIRT_L_CONTINEN_10005_20150703_21000101.DBL> is already uncompressed (.DBL.DIR directory already exists).
2020-12-20T19:57:07.523317 13fa8c657432 maja-processing-3.2.2 3.2 [000000170640] [P] Starting GIP_L2DIFT uncompressing ...
2020-12-20T19:57:07.524152 13fa8c657432 maja-processing-3.2.2 3.2 [000000170640] [I] UnCompress the DBL product </root/work/34SFJ/fix/in/S2A_TEST_GIP_L2DIFT_L_CONTINEN_10005_20150703_21000101.DBL> ...
2020-12-20T19:57:07.524354 13fa8c657432 maja-processing-3.2.2 3.2 [000000170640] [I] The DBL product </root/work/34SFJ/fix/in/S2A_TEST_GIP_L2DIFT_L_CONTINEN_10005_20150703_21000101.DBL> is already uncompressed (.DBL.DIR directory already exists).
2020-12-20T19:57:07.524539 13fa8c657432 maja-processing-3.2.2 3.2 [000000170640] [P] Starting GIP_L2ALBD uncompressing ...
2020-12-20T19:57:07.525347 13fa8c657432 maja-processing-3.2.2 3.2 [000000170640] [I] UnCompress the DBL product </root/work/34SFJ/fix/in/S2A_TEST_GIP_L2ALBD_L_CONTINEN_10005_20150703_21000101.DBL> ...
2020-12-20T19:57:07.525537 13fa8c657432 maja-processing-3.2.2 3.2 [000000170640] [I] The DBL product </root/work/34SFJ/fix/in/S2A_TEST_GIP_L2ALBD_L_CONTINEN_10005_20150703_21000101.DBL> is already uncompressed (.DBL.DIR directory already exists).
2020-12-20T19:57:07.525750 13fa8c657432 maja-processing-3.2.2 3.2 [000000170640] [P] Starting GIP_L2DIRT reading ...
2020-12-20T19:57:07.527243 13fa8c657432 maja-processing-3.2.2 3.2 [000000170640] [P] Starting GIP_L2DIFT reading ...
2020-12-20T19:57:07.528683 13fa8c657432 maja-processing-3.2.2 3.2 [000000170640] [P] Starting GIP_L2ALBD reading ...
2020-12-20T19:57:07.530141 13fa8c657432 maja-processing-3.2.2 3.2 [000000170640] [P] Starting GIP_L2TOCR reading ...
2020-12-20T19:57:07.626975 13fa8c657432 maja-processing-3.2.2 3.2 [000000170640] [I] Initialize the Plugin instance detected with the Plugin Name 'SENTINEL2_TM' ...
2020-12-20T19:57:07.636785 13fa8c657432 maja-processing-3.2.2 3.2 [000000170640] [I] The number of streaming lines of buffer for reading/writing images is '800'.
2020-12-20T19:57:07.637023 13fa8c657432 maja-processing-3.2.2 3.2 [000000170640] [I] Using default Admin configuration camera file.
2020-12-20T19:57:07.645130 13fa8c657432 maja-processing-3.2.2 3.2 [000000170640] [I] Starting reading data for the satellite 'SENTINEL-2B' and the plugin used is 'SENTINEL2_TM' ...
2020-12-20T19:57:07.645401 13fa8c657432 maja-processing-3.2.2 3.2 [000000170640] [P] Starting GIP_L2SMAC reading ...
2020-12-20T19:57:07.647284 13fa8c657432 maja-processing-3.2.2 3.2 [000000170640] [P] Starting GIP_L2TOCR reading ...
2020-12-20T19:57:07.648436 13fa8c657432 maja-processing-3.2.2 3.2 [000000170640] [I] UnCompress the DBL product </root/work/34SFJ/fix/in/S2B_TEST_GIP_L2TOCR_L_CONTINEN_10002_20150703_21000101.DBL> ...
2020-12-20T19:57:07.648742 13fa8c657432 maja-processing-3.2.2 3.2 [000000170640] [I] The DBL product </root/work/34SFJ/fix/in/S2B_TEST_GIP_L2TOCR_L_CONTINEN_10002_20150703_21000101.DBL> is already uncompressed (.DBL.DIR directory already exists).
2020-12-20T19:57:07.650226 13fa8c657432 maja-processing-3.2.2 3.2 [000000170640] [P] Starting GIP_L2WATV reading ...
2020-12-20T19:57:07.651349 13fa8c657432 maja-processing-3.2.2 3.2 [000000170640] [I] UnCompress the DBL product </root/work/34SFJ/fix/in/S2B_TEST_GIP_L2WATV_L_CONTINEN_10005_20150703_21000101.DBL> ...
2020-12-20T19:57:07.651636 13fa8c657432 maja-processing-3.2.2 3.2 [000000170640] [I] The DBL product </root/work/34SFJ/fix/in/S2B_TEST_GIP_L2WATV_L_CONTINEN_10005_20150703_21000101.DBL> is already uncompressed (.DBL.DIR directory already exists).
2020-12-20T19:57:07.652115 13fa8c657432 maja-processing-3.2.2 3.2 [000000170640] [I] Allocating LookUpTable image dimension = 3
2020-12-20T19:57:07.652475 13fa8c657432 maja-processing-3.2.2 3.2 [000000170640] [P] Starting GIP_L2DIRT uncompressing ...
2020-12-20T19:57:07.653636 13fa8c657432 maja-processing-3.2.2 3.2 [000000170640] [I] UnCompress the DBL product </root/work/34SFJ/fix/in/S2B_TEST_GIP_L2DIRT_L_CONTINEN_10002_20150703_21000101.DBL> ...
2020-12-20T19:57:07.653928 13fa8c657432 maja-processing-3.2.2 3.2 [000000170640] [I] The DBL product </root/work/34SFJ/fix/in/S2B_TEST_GIP_L2DIRT_L_CONTINEN_10002_20150703_21000101.DBL> is already uncompressed (.DBL.DIR directory already exists).
2020-12-20T19:57:07.654197 13fa8c657432 maja-processing-3.2.2 3.2 [000000170640] [P] Starting GIP_L2DIFT uncompressing ...
2020-12-20T19:57:07.655326 13fa8c657432 maja-processing-3.2.2 3.2 [000000170640] [I] UnCompress the DBL product </root/work/34SFJ/fix/in/S2B_TEST_GIP_L2DIFT_L_CONTINEN_10002_20150703_21000101.DBL> ...
2020-12-20T19:57:07.655621 13fa8c657432 maja-processing-3.2.2 3.2 [000000170640] [I] The DBL product </root/work/34SFJ/fix/in/S2B_TEST_GIP_L2DIFT_L_CONTINEN_10002_20150703_21000101.DBL> is already uncompressed (.DBL.DIR directory already exists).
2020-12-20T19:57:07.655913 13fa8c657432 maja-processing-3.2.2 3.2 [000000170640] [P] Starting GIP_L2ALBD uncompressing ...
2020-12-20T19:57:07.656710 13fa8c657432 maja-processing-3.2.2 3.2 [000000170640] [I] UnCompress the DBL product </root/work/34SFJ/fix/in/S2B_TEST_GIP_L2ALBD_L_CONTINEN_10003_20150703_21000101.DBL> ...
2020-12-20T19:57:07.656894 13fa8c657432 maja-processing-3.2.2 3.2 [000000170640] [I] The DBL product </root/work/34SFJ/fix/in/S2B_TEST_GIP_L2ALBD_L_CONTINEN_10003_20150703_21000101.DBL> is already uncompressed (.DBL.DIR directory already exists).
2020-12-20T19:57:07.657133 13fa8c657432 maja-processing-3.2.2 3.2 [000000170640] [P] Starting GIP_L2DIRT reading ...
2020-12-20T19:57:07.658446 13fa8c657432 maja-processing-3.2.2 3.2 [000000170640] [P] Starting GIP_L2DIFT reading ...
2020-12-20T19:57:07.659776 13fa8c657432 maja-processing-3.2.2 3.2 [000000170640] [P] Starting GIP_L2ALBD reading ...
2020-12-20T19:57:07.661057 13fa8c657432 maja-processing-3.2.2 3.2 [000000170640] [P] Starting GIP_L2TOCR reading ...
2020-12-20T19:57:07.756924 13fa8c657432 maja-processing-3.2.2 3.2 [000000170640] [P] Starting L2Processor::PreProcessing() done.
2020-12-20T19:57:07.757683 13fa8c657432 maja-processing-3.2.2 3.2 [000000170640] [I] L2BackwardProcessor::PreProcessing() done.
2020-12-20T19:57:07.757839 13fa8c657432 maja-processing-3.2.2 3.2 [000000170640] [I] Starting L2BackwardProcessor::ScientificProcessing() ...
2020-12-20T19:57:07.767404 13fa8c657432 maja-processing-3.2.2 3.2 [000000170640] [I] Initialize the Plugin instance detected with the Plugin Name 'SENTINEL2_TM' ...
2020-12-20T19:57:07.779693 13fa8c657432 maja-processing-3.2.2 3.2 [000000170640] [I] The number of streaming lines of buffer for reading/writing images is '800'.
2020-12-20T19:57:07.779919 13fa8c657432 maja-processing-3.2.2 3.2 [000000170640] [I] Using default Admin configuration camera file.
2020-12-20T19:57:07.806445 13fa8c657432 maja-processing-3.2.2 3.2 [000000170640] [I] Start Sentinel2 L1 ImageFileReader ...
2020-12-20T19:59:42.308856 13fa8c657432 maja-processing-3.2.2 3.2 [000000170640] [W] Sentinel2L1ImageFileReaderBase::GetDetFooIndexShift: The detector indices are not in ascending order or do not have a constant shift with the feature indices in the MSK_DETFOO /root/work/34SFJ/fix/in/S2A_MSIL1C_20190330T092031_N0207_R093_T34SFJ_20190330T105736.SAFE/GRANULE/L1C_T34SFJ_A019681_20190330T092028/../../GRANULE/L1C_T34SFJ_A019681_20190330T092028/QI_DATA/MSK_DETFOO_B03.gml !
2020-12-20T19:59:43.314864 13fa8c657432 maja-processing-3.2.2 3.2 [000000170640] [W] Sentinel2L1ImageFileReaderBase::GetDetFooIndexShift: The detector indices are not in ascending order or do not have a constant shift with the feature indices in the MSK_DETFOO /root/work/34SFJ/fix/in/S2A_MSIL1C_20190330T092031_N0207_R093_T34SFJ_20190330T105736.SAFE/GRANULE/L1C_T34SFJ_A019681_20190330T092028/../../GRANULE/L1C_T34SFJ_A019681_20190330T092028/QI_DATA/MSK_DETFOO_B04.gml !
2020-12-20T19:59:45.309442 13fa8c657432 maja-processing-3.2.2 3.2 [000000170640] [W] Sentinel2L1ImageFileReaderBase::GetDetFooIndexShift: The detector indices are not in ascending order or do not have a constant shift with the feature indices in the MSK_DETFOO /root/work/34SFJ/fix/in/S2A_MSIL1C_20190330T092031_N0207_R093_T34SFJ_20190330T105736.SAFE/GRANULE/L1C_T34SFJ_A019681_20190330T092028/../../GRANULE/L1C_T34SFJ_A019681_20190330T092028/QI_DATA/MSK_DETFOO_B05.gml !
2020-12-20T19:59:45.590034 13fa8c657432 maja-processing-3.2.2 3.2 [000000170640] [W] Sentinel2L1ImageFileReaderBase::GetDetFooIndexShift: The detector indices are not in ascending order or do not have a constant shift with the feature indices in the MSK_DETFOO /root/work/34SFJ/fix/in/S2A_MSIL1C_20190330T092031_N0207_R093_T34SFJ_20190330T105736.SAFE/GRANULE/L1C_T34SFJ_A019681_20190330T092028/../../GRANULE/L1C_T34SFJ_A019681_20190330T092028/QI_DATA/MSK_DETFOO_B06.gml !
2020-12-20T19:59:45.865966 13fa8c657432 maja-processing-3.2.2 3.2 [000000170640] [W] Sentinel2L1ImageFileReaderBase::GetDetFooIndexShift: The detector indices are not in ascending order or do not have a constant shift with the feature indices in the MSK_DETFOO /root/work/34SFJ/fix/in/S2A_MSIL1C_20190330T092031_N0207_R093_T34SFJ_20190330T105736.SAFE/GRANULE/L1C_T34SFJ_A019681_20190330T092028/../../GRANULE/L1C_T34SFJ_A019681_20190330T092028/QI_DATA/MSK_DETFOO_B07.gml !
2020-12-20T19:59:46.144844 13fa8c657432 maja-processing-3.2.2 3.2 [000000170640] [W] Sentinel2L1ImageFileReaderBase::GetDetFooIndexShift: The detector indices are not in ascending order or do not have a constant shift with the feature indices in the MSK_DETFOO /root/work/34SFJ/fix/in/S2A_MSIL1C_20190330T092031_N0207_R093_T34SFJ_20190330T105736.SAFE/GRANULE/L1C_T34SFJ_A019681_20190330T092028/../../GRANULE/L1C_T34SFJ_A019681_20190330T092028/QI_DATA/MSK_DETFOO_B8A.gml !
2020-12-20T19:59:46.422698 13fa8c657432 maja-processing-3.2.2 3.2 [000000170640] [W] Sentinel2L1ImageFileReaderBase::GetDetFooIndexShift: The detector indices are not in ascending order or do not have a constant shift with the feature indices in the MSK_DETFOO /root/work/34SFJ/fix/in/S2A_MSIL1C_20190330T092031_N0207_R093_T34SFJ_20190330T105736.SAFE/GRANULE/L1C_T34SFJ_A019681_20190330T092028/../../GRANULE/L1C_T34SFJ_A019681_20190330T092028/QI_DATA/MSK_DETFOO_B11.gml !
2020-12-20T19:59:46.990333 13fa8c657432 maja-processing-3.2.2 3.2 [000000170640] [W] Sentinel2L1ImageFileReaderBase::GetDetFooIndexShift: The detector indices are not in ascending order or do not have a constant shift with the feature indices in the MSK_DETFOO /root/work/34SFJ/fix/in/S2A_MSIL1C_20190330T092031_N0207_R093_T34SFJ_20190330T105736.SAFE/GRANULE/L1C_T34SFJ_A019681_20190330T092028/../../GRANULE/L1C_T34SFJ_A019681_20190330T092028/QI_DATA/MSK_DETFOO_B01.gml !
2020-12-20T19:59:47.061955 13fa8c657432 maja-processing-3.2.2 3.2 [000000170640] [E] ******************************************************************************************************  [vnsMajaMainProcessor.cxx:main:130]
2020-12-20T19:59:47.062295 13fa8c657432 maja-processing-3.2.2 3.2 [000000170640] [E] ScientificProcessing: ITK Exception: itk::ERROR: PersistentStreamingMaskConditionalStatisticsVectorImageFilter(0x6d7f3a0): Inputs do not occupy the same physical space! 
InputImage Origin: [6.0006000e+05, 4.3999800e+06], InputImage_1 Origin: [6.0012000e+05, 4.3999200e+06]
	Tolerance: 1.2000000e-04
InputImage Spacing: [1.2000000e+02, -1.2000000e+02], InputImage_1 Spacing: [2.4000000e+02, -2.4000000e+02]
	Tolerance: 1.2000000e-04
 [itkImageToImageFilter.hxx:248]  [vnsMajaMainProcessor.cxx:main:130]
2020-12-20T19:59:47.062600 13fa8c657432 maja-processing-3.2.2 3.2 [000000170640] [E]   [vnsMajaMainProcessor.cxx:main:130]
2020-12-20T19:59:47.062873 13fa8c657432 maja-processing-3.2.2 3.2 [000000170640] [E] ******************************************************************************************************  [vnsMajaMainProcessor.cxx:main:130]
```

---

So, is there a version of StartMaja that we can use to drive MAJA v3.2.2? Was this possible at some point in time and can we recreate it? Is the above towards such a direction or is it pointless to be trying that?

Thank you once more for these tools and you time.

[ostartmaja]: https://github.com/olivierhagolle/Start_maja
[gipp]: http://tully.ups-tlse.fr/olivier/gipp

**Comment by petket-5** [comment]
Hi,

StartMaja is indeed still backwards compatible. The error you are seeing is due to another problem: Maja 3.2.2 is not able to mix different L2Coarse resolutions:

```
InputImage Spacing: [1.2000000e+02, -1.2000000e+02], InputImage_1 Spacing: [2.4000000e+02, -2.4000000e+02]
```

This means that one image is at 120m and the other at 240m resolution. We introduced the 120m as a default in Maja 4.2 as it improves the overall quality of the product, however it is also possible to still process it in 240m. For that, you have to use either Maja 4.2 or 3.2.2 and restart the time-series processing by verifying that the following parameter in the files `userconf/MAJAUserConfig_SENTINEL2*xml`:
```
            <L2CoarseResolution>240</L2CoarseResolution>
```

Kind regards,
Peter
